zarr/storage.py

```diff
@@ -1114,9 +1117,26 @@ def dir_path(self, path=None):
     def listdir(self, path=None):
         dir_path = self.dir_path(path)
         try:
-            out = sorted(p.rstrip('/').rsplit('/', 1)[-1]
-                         for p in self.fs.ls(dir_path, detail=False))
-            return out
+            children = sorted(p.rstrip('/').rsplit('/', 1)[-1]
+                              for p in self.fs.ls(dir_path, detail=False))
+            if self.key_separator == '/':
```

Comment:
Maybe simpler to invert the test and return `children` here, being closer to the original and the more common path

Reply:
good idea! implemented in babd76a78b53277c98917051a0cfbbc8728f0452